### PR TITLE
my-services tutorial and .dockstore.yml template

### PR DESCRIPTION
**Pull request by NatalieEO**
dockstore/dockstore#2709

### CODE REVIEW

File: docs/getting-started/getting-started-with-services.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+Next, is a required key named ``service`` where your service description will lie.
+
+Within this, you should specify the type which can be docker-compose, swarm, kubernetes, or helm.
+
+.. Not actually working right now
+
+Following lines 4-9, it is good practice to include the name, author, and a description of your service. This information will be parsed and displayed on your service's page on Dockstore.
+
+Next, is the files key. Here you can specify a list of other files (like scripts, READmes, and test parameter files) for Dockstore to index that are contained in your repository and are needed for your service.
```
Comment (garyluu) [inline docs/getting-started/getting-started-with-services.rst]: READMEs

File: docs/getting-started/getting-started-with-services.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+Following that is a scripts section. Here you denote the script files or commands to be used for steps like starting, provisioning, specifying the port, and stopping for your service. Other possible keys not listed above include preprovision, prestart, and healthcheck.
+
+Next is a section where you can list any environment variables a service needs to be passed into its scripts by the launcher. In this example, we provide a default HTTP port.
+
+Finally, if your service needs to have data provisioned locally, you should include a data section in your ``dockstore.yml``.
```
Comment (garyluu) [inline docs/getting-started/getting-started-with-services.rst]: `.dockstore.yml`

File: docs/getting-started/getting-started-with-services.rst [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+
+Registering Your Service
+------------------------
+Registering your service works differently than tools and workflows on Dockstore. For services, you need to install our Dockstore Github application into at least one of your organizations.
```
Comment (garyluu) [inline docs/getting-started/getting-started-with-services.rst]: GitHub everywhere

File: docs/getting-started/getting-started-with-services.rst [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+On this page, you will need to click the + icon located on the top left. This is the “Manage on GitHub” button, and will redirect you to the screen pictured below in order to install our Dockstore GitHub app into an organization of your choice.
+
+.. figure:: /assets/images/docs/install-dockstore-app.png
+
+After choosing an organization to install Dockstore on, you will be taken to the following GitHub page.
+
+.. figure:: /assets/images/docs/github-app-settings-page.png
+
+Here you can select whether to give access to all repositories or only select ones. If the organization you choose is intended to be just for services, we recommend choosing all repositories. Otherwise, it is probably more intuitive to select only certain repositories. Click save and you will be taken back to the my services page on Dockstore.
```
Comment (garyluu) [inline docs/getting-started/getting-started-with-services.rst]: "My Services" everywhere

File: docs/assets/templates/.dockstore.yml [hunk 5]
```diff
@@ -0,0 +1,65 @@
+# The dockstore.yml version; 1.1 is required for services
```
Comment (garyluu) [inline docs/assets/templates/.dockstore.yml]: .dockstore.yml

File: docs/getting-started/getting-started-with-services.rst [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+- Get familiarized with the concept of services on Dockstore
+- Learn the basics of creating a service using a ``.dockstore.yml`` file
+- Learn how to register a service and create versions for it
+- Publish your service to Dockstore
+
+.. David will add context here.
+
+Services are meant to be long running processes, usually web services or interactive applications, that can be launched by a user. The entire process for creating a service that can launch successfully is complex and dependent on the service or application you are trying to describe; so it is beyond the scope of this tutorial.
+However, this document will outline what is needed to register, update, and publish a service onto Dockstore. Also, please keep in mind that service functionality is currently in beta mode.
```
Comment (agduncan94) [inline docs/getting-started/getting-started-with-services.rst]: I would add a note box to the top (see https://beta.docs.dockstore.org/en/develop/) that says that the feature is in beta mode.

https://sphinx-rtd-theme.readthedocs.io/en/stable/demo/demo.html#admonitions

[the remark this answers is not in the document]
Comment (agduncan94) [reply docs/getting-started/getting-started-with-services.rst]: @denis-yuen I posted above, I suggested something like https://sphinx-rtd-theme.readthedocs.io/en/stable/demo/demo.html#admonitions

File: docs/getting-started/getting-started-with-services.rst [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+Publishing Your Service
+-----------------------
+When you select a repository from this panel to view, you will notice that information is missing. Clicking the Versions tab, you will see an info box letting you know that this service does not have any versions. When your service is in this state, the publish button is disabled.
+
+To add a version, get the information to populate, and gain the ability to publish, all you need to do is create a release or prerelease on GitHub on a repository that has a valid ``.dockstore.yml`` located in its root directory. Any time this is done, Dockstore will automatically create versions for you.
+
+.. figure:: /assets/images/docs/services-versions-tab.png
+
+Now the Files tab will have your ``.dockstore.yml`` under the Primary Descriptor. Additionally, the files you choose to list in the in the file section of the ``.dockstore.yml`` will also be visible in the Files tab.
```
Comment (coverbeck) [inline docs/getting-started/getting-started-with-services.rst]: `in the files section` (missing s)

File: docs/getting-started/getting-started-with-services.rst [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+Services are meant to be long running processes, usually web services or interactive applications, that can be launched by a user. The entire process for creating a service that can launch successfully is complex and dependent on the service or application you are trying to describe; so it is beyond the scope of this tutorial.
+However, this document will outline what is needed to register, update, and publish a service onto Dockstore. Also, please keep in mind that service functionality is currently in beta mode.
+
+Create Your Service
+-------------------
+
+The first step is to create a file named ``.dockstore.yml``. The ``.dockstore.yml``  is a configuration file used to describe your service. Technically, having a valid YAML and the right version is
+all that is required for your service to appear on Dockstore. But, we will briefly cover an example ``.dockstore.yml`` written for the Xena Hub to show what will generally be needed to create a working service.
+You can also view a template ``.dockstore.yml`` file `here
+<https://github.com/dockstore/dockstore-documentation/tree/develop/docs/assets/templates/.dockstore.yml>`_.
```
[the remark this answers is not in the document]
Comment (NatalieEO) [reply docs/getting-started/getting-started-with-services.rst]: put up in the note at the top

**Pull request merged**